io/module_write_netcdf_parallel.F90

```diff
@@ -378,15 +378,20 @@ subroutine write_netcdf_parallel(fieldbundle, wrtfb, filename, mpi_comm, mype, i
               ! rescaling (plus it allows the ability to adjust the packing
               ! range)
               scale_fact = (dataMax - dataMin) / (2**nbits-1); offset = dataMin
-              allocate(arrayr4_3d_save(i1:i2,j1:j2,k1:k2))
-              arrayr4_3d_save(i1:i2,j1:j2,k1:k2)=arrayr4_3d(i1:i2,j1:j2,k1:k2)
-              arrayr4_3d = scale_fact*(nint((arrayr4_3d_save - offset) / scale_fact)) + offset
-              ! compute max abs compression error.
-              compress_err(i) = &
-              maxval(abs(arrayr4_3d_save(i1:i2,j1:j2,k1:k2)-arrayr4_3d(i1:i2,j1:j2,k1:k2)))
-              deallocate(arrayr4_3d_save)
-              call mpi_allreduce(mpi_in_place,compress_err(i),1,mpi_real4,mpi_max,mpi_comm,ierr)
-              !print *,'field name=',trim(fldName),dataMin,dataMax,compress_err(i)
+              if (scale_fact > 0.) then
```

Comment:
Is original code causing issues or the code changes are to skip the allocating array and mpi calls for constant field?

Reply:
This change is meant to address an issue in the original code.

Reply:
Do you know what is the issue?

Reply:
It was a divide by zero if `scale_fact = 0`